pkg/config/shared.go

```diff
@@ -118,14 +118,20 @@ type HTTP struct {
 
 // Redis represents Redis configuration.
 type Redis struct {
-	Address   string   `name:"address" description:"Address of the Redis server"`
-	Password  string   `name:"password" description:"Password of the Redis server"`
-	Database  int      `name:"database" description:"Redis database to use"`
-	Namespace []string `name:"namespace" description:"Namespace for Redis keys"`
+	Address            string   `name:"address" description:"Address of the Redis server"`
+	Password           string   `name:"password" description:"Password of the Redis server"`
+	Database           int      `name:"database" description:"Redis database to use"`
+	Namespace          []string `name:"namespace" description:"Namespace for Redis keys"`
+	EnableFailover     bool     `name:"enable-failover" description:"Enable failover using Redis Sentinel (default false)"`
+	FailoverAddresses  []string `name:"failover-addresses" description:"Redis Sentinel server addresses (when enable-failover=true)"`
+	FailoverMasterName string   `name:"failover-master-name" description:"Redis Sentinel master name (when enable-failover=true)"`
 }
 
 // IsZero returns whether the Redis configuration is empty.
-func (r Redis) IsZero() bool { return r.Address == "" && r.Database == 0 && len(r.Namespace) == 0 }
+func (r Redis) IsZero() bool {
+	return (r.EnableFailover && r.FailoverMasterName == "" && len(r.FailoverAddresses) == 0) ||
```

Comment:
This is not correct - `r.Namespace` and `r.Database` are only checked with `!r.EnableFailover` in proposed implementation, but they have to be checked always

Reply:
Yeah, totally missed that. Should be OK now